File: Strings/2108. Find First Palindromic String in the Array.java
public class Solution {
        public static String firstPalindrome(String[] words) {
        for (String word : words) {
            StringBuilder reverse = new StringBuilder(word).reverse();
                if (word.equals(reverse.toString())) {
                     return word;
                    }
                 }
        return ""; 
         }                                                                                   
}

---------------------------------------------------------------------------------------------------------------------------------

class Solution {
    public String firstPalindrome(String[] words) {
        for(String s:words){
            if(checkPallindrome(s)){
                return s;
            }
        }
        return "";
    }

    private boolean checkPallindrome(String s){
        int left=0;
        int right=s.length()-1;

        while(left<=right){
            if(s.charAt(left)==s.charAt(right)){
                left++;
                right--;
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: Strings/2108. Find First Palindromic String in the Array.py
class Solution:
    def firstPalindrome(self, words):
        for s in words:
            if self.checkPalindrome(s):
                return s
        return ""

    def checkPalindrome(self, s):
        left = 0
        right = len(s) - 1

        while left <= right:
            if s[left] == s[right]:
                left += 1
                right -= 1
            else:
                return False
        return True
